### DISCUSSION TITLE
Why is it very slow to load and enter after entering the password? How to optimize it?

### DISCUSSION THREAD

**Discussion by windkwbs** [open-webui/open-webui#3051] [General]
Why is it very slow to load and enter after entering the password? How to optimize it?

**Comment by JuanjoJG-WATA** [comment]
It's maybe on your end. How much RAM do you have? It goes pretty fluid on my server with 16GB

**Reply by windkwbs** [reply]
Download a new browser and try to load it. The loading will be very slow. If you think it is faster, it may be cached before.

**Reply by JuanjoJG-WATA** [reply]
i don't know, to me it's still pretty fast. I downloaded Safari and tried it. Deleted the cache there and on Chrome and tried it again, same result.

https://github.com/open-webui/open-webui/assets/165764759/2011b37f-57bc-4843-bba2-789154e1273a

**Reply by windkwbs** [reply]
Thank you. It seems my equipment is old.

**Reply by turbosnail11** [reply]
Checking the javascript console, it seems "models" takes a long time to load.  Not that big (16K) but it took 134 seconds to load.  I do have a lot of models, is it pinging each one to make sure they work?

This is the problematic page:

http://your_server:3000/api/models

**Comment by amirasyraf** [comment]
For me at least, it seems there is some sort of "cold start" performance issue, i.e. if I haven't used open-webui for several hours to a day, launching it the first time is extremely slow. Takes 10+ seconds. Subsequent launches are quick (even with cache disabled).

Looking at the stats, CPU and memory climbs, as if it's just booting up and loading. 

But the thing is, I'm not using any local models. Every single thing has been set to OpenAI, so there should be no models getting loaded to contribute to the slowness.

Memory usage is rather low as expected at 250 - 300 MB.

Is there a way to keep the instance "warm/hot" or something? As I've configured it now, with 100% OpenAI, it's just a UI so I don't understand why it would be so slow to load.

Hosted on EC2 t4g.small which has 2 vCPU and 2 GB memory, and I've configured 2 GB of swap.

**Reply by TedLin1993** [reply]
My open-webui memory usage is around 600MB. Could you tell me what's your open-webui version and environment variables?

**Reply by turbosnail11** [reply]
I figured out my issue, it was the models (http://your_server:3000/api/models). 

I have 10 api servers on there but one of them is my home PC which is often asleep.  Open WebUI seems to wait 130 seconds until it times out?  The odd thing is that even though it times out, the model is still available in the model listing - so not sure what the point of that is...  Disabling the the api on my home pc resolved the issue for me.

**Comment by olekgolus11** [comment]
Yeah, I run it on Google Cloud Run with 8GiB and 4vCPU but it takes so long to load, not only on the start, but navigation through options and conversation feels very slow :/

**Comment by afonsonogsousa** [comment]
I found the solution.
Simply remove the api link for Ollama from your connections in admin settings. Default is some localhost port, just remove it. That solved it for me. Previously taking 2min to load, currently less than 5 seconds. Deployed on cloud using digitalOcean, 1gb + 1cpu is more than enough for it to run using APIs only, no local models.
You're welcome :)

**Reply by teddythinh** [reply]
Do you mean this connection? `http://host.docker.internal:11434`

![image](https://github.com/user-attachments/assets/abafbfa7-632d-426a-9d4f-dbfe01d27e04)

**Reply by afonsonogsousa** [reply]
yes

**Reply by mejiacm** [reply]
> I found the solution. Simply remove the api link for Ollama from your connections in admin settings. Default is some localhost port, just remove it. That solved it for me. Previously taking 2min to load, currently less than 5 seconds. Deployed on cloud using digitalOcean, 1gb + 1cpu is more than enough for it to run using APIs only, no local models. You're welcome :)

I can confirm this was the trick, I disabled the link as you suggested and got it working the way it should. 

Thank you!!!

**Reply by abdelilahakebli** [reply]
i confirm,  and use only OpenAI Api and this did the trick for me also.

Thank you

**Reply by Hussein-Ali-6** [reply]
Thanks a lot, worked for me.

in Google Dev Tools, I noticed there is a request for fetching models, which blocks the entire app, take too much time.
This solved the issue

**Reply by anthonyg45157** [reply]
Disabling ollama worked for me as well, I only use open AI API anyhow, thank you for this

**Comment by okaeiz** [comment]
I have disabled Ollama but still am experiencing annoying performance issues. 

I read it somewhere that disabling the search query generation for both RAG and Web can increase the chat loading speed and I did disable them both but the issue persists.

And the problem is not only with the login page and chats; it seems like the entire application is extremely slow at times.

I'd be grateful if anyone share their solutions if they come up to any.

**Reply by charles-le-ponzi** [reply]
I suspect it's related to the database for some reason. Is your database huge? mine is only about 5gb, but it takes a whole 3-4 minutes to open the web ui.... on a nvme ssd. I also noticed one of the gpus is getting used every time I reload the page or click somewhere. I run docker with cuda enabled. I used to run the pip install without cuda on another machine and that loaded in about 2mins.

**Reply by galvanoid** [reply]
I suspect same. My webui.db is about 6gb and owui practically unusable. If I remove chats, collections, etc.,  the database is reduced and everything is going as it should.

**Comment by galvanoid** [comment]
I think I've identified the specific problem. My "file" table has 153,644 records.
If I try to open something in owui that looks at those records, it takes ages.
For example, if I load an old chat that hasn't used knowledge (a conversational question), it loads it instantly.
But if I load a chat that has used knowledge, it takes minutes.
Likewise, if I go to knowledge, it takes ages to load my knowledge list.

**Comment by rgaricano** [comment]
a way to speedy is caching some routes, usefull for a stable list of models:
It is quite noticeable, but of course not in dynamic calls to the database.

If sqlitedb, enabling  WAL can improve it: https://github.com/open-webui/open-webui/discussions/15123

Caching, (from https://linux.do/t/topic/322664 it was posted here https://github.com/open-webui/open-webui/discussions/14609#discussioncomment-13572979 )

cache models
```
# models cache
location /api/models {
    proxy_cache nginx_cache; # 这里的 nginx_cache 就是我设定的 proxy_cache_path
    proxy_cache_key $request_uri;

    # cache for 30 minutes
    proxy_cache_valid 200 30m;
    # make nignx auto update cache content
    proxy_cache_background_update on; # 这里要求 nginx 自动更新 api
    proxy_cache_use_stale updating;
    proxy_cache_revalidate on;
    proxy_cache_min_uses 1;

    proxy_pass http://127.0.0.1:8888;
    add_header X-Cache-Status $upstream_cache_status;
}
```

cache statics
```
# Static resources
location ~* \.(css|js|jpg|jpeg|png|gif|ico|svg|woff|woff2|ttf|eot)(.*) {
    proxy_pass http://127.0.0.1:8888;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

    # Browser cache
    expires 1d;
    add_header Cache-Control "public, no-transform";
}
```

e.g.
```
http {
    include       mime.types;
    default_type  application/octet-stream;

    # 我只缓存 /api/models, 用到的空间较少
	proxy_cache_path /var/cache/nginx 
		levels=1:2
		keys_zone=nginx_cache:1m
		max_size=100m
		inactive=24h
		use_temp_path=off;
    keepalive_timeout  65;


	server {
		listen   80;
		server_name openwebui.local;

		location / {
			proxy_pass http://127.0.0.1:8888/;

			proxy_set_header X-Real-IP $remote_addr;
			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
			proxy_set_header Host $host;

			# websocket header
			proxy_http_version 1.1;
			proxy_set_header Upgrade $http_upgrade;
			proxy_set_header Connection 'upgrade';
			# Timeouts for WebSocket connections
			proxy_read_timeout 60s;
			proxy_send_timeout 60s;
		}

		# Static resources
		location ~* \.(css|js|jpg|jpeg|png|gif|ico|svg|woff|woff2|ttf|eot)(.*) {
			proxy_pass http://127.0.0.1:8888;
			proxy_set_header Host $host;
			proxy_set_header X-Real-IP $remote_addr;
			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

			# Browser cache
			expires 1d;
			add_header Cache-Control "public, no-transform";
		}


		# models
		location /api/models {
			proxy_cache nginx_cache;
			proxy_cache_key $request_uri;

			# cache for 30 minutes
			proxy_cache_valid 200 30m;
			# make nignx auto update cache content
			proxy_cache_background_update on;
			proxy_cache_use_stale updating;
			proxy_cache_revalidate on;
			proxy_cache_min_uses 1;

			proxy_pass http://127.0.0.1:8888;
			add_header X-Cache-Status $upstream_cache_status;
		}
	}

}
```

**Reply by galvanoid** [reply]
Confirmed, enabling wal and chache improves a lot.


Now I'm going to try to move the database to Postgres: https://github.com/taylorwilsdon/open-webui-postgres-migration?tab=readme-ov-file

Regards.

**Reply by rgaricano** [reply]
commit for WAL: https://github.com/tcx4c70/open-webui/commit/9f2ad8168cd1a1e579ea099866928b4a054f9188
